src/gfx/qt.cpp

```diff
@@ -586,10 +614,175 @@ const char *GfxProcQT::supportedformatsQT()
         formatstring->resize(formatstring->size() - 1);
         formatstring->append(supportedformatsFfmpeg());
 #endif
+
+#ifdef HAVE_LIBRAW
+        formatstring->resize(formatstring->size() - 1);
+        formatstring->append(supportedformatsLibraw());
+#endif
     }
     return formatstring->constData();
 }
 
+#ifdef HAVE_LIBRAW
+
+const char *GfxProcQT::supportedformatsLibraw()
+{
+    return ".3fr.arw.cr2.crw.ciff.cs1.dcr.dng.erf.iiq.k25.kdc.mef.mos.mrw.nef.nrw.orf.pef.raf.raw.rw2.rwl.sr2.srf.srw.x3f.";
+}
+
+QImageReader *GfxProcQT::readbitmapLibraw(int &w, int &h, int &orientation, QString imagePath)
+{
+    LibRaw libRaw;
+    int ret = libRaw.open_file(imagePath.toUtf8().constData());
+    if (ret > 0 || LIBRAW_FATAL_ERROR(ret)
+            || libRaw.imgdata.sizes.width <= 0
+            || libRaw.imgdata.sizes.height <= 0)
+    {
+        LOG_debug << "Unreadable RAW image";
+        return NULL;
+    }
+
+    libraw_processed_image_t *output = NULL;
+    const libraw_data_t &imgdata = libRaw.imgdata;
+    LOG_debug << "Processing RAW image: " << imagePath.toUtf8().constData()
+              << " " << imgdata.sizes.width
+              << " " << imgdata.sizes.height
+              << " " << imgdata.thumbnail.twidth
+              << " " << imgdata.thumbnail.theight
+              << " " << imgdata.sizes.flip;
+
+    if (imgdata.thumbnail.twidth > 0 && imgdata.thumbnail.theight > 0)
```

Comment:
Should we consider discarding libraw's preview image if it is too small?

Reply:
I did it in my first approach, but I changed it to follow the same approach as the webclient with these images (always use the thumbnail if available).